server/middlewares/csp.ts

```diff
@@ -0,0 +1,42 @@
+import * as helmet from 'helmet'
+
+const baseDirectives = {
+  defaultSrc: ["'none'"], // by default, not specifying default-src = '*'
+  connectSrc: ['*', 'data:'],
+  mediaSrc: ["'self'", 'blob:'],
+  fontSrc: ["'self'", 'data:'],
+  imgSrc: ["'self'", 'data:'],
+  scriptSrc: ["'self' 'unsafe-inline'"],
+  styleSrc: ["'self' 'unsafe-inline'"],
+  // objectSrc: ["'none'"], // only define to allow plugins, else let defaultSrc 'none' block it
+  formAction: ["'self'"],
+  frameAncestors: ["'none'"],
+  baseUri: ["'self'"],
+  pluginTypes: ["'none'"],
+  manifestSrc: ["'self'"],
+  frameSrc: ["'self'"], // instead of deprecated child-src / self because of test-embed
+  workerSrc: ["'self'"], // instead of deprecated child-src
+  upgradeInsecureRequests: true,
+  blockAllMixedContent: true
```

Comment:
This line is not needed, because `upgrade-insecure-requests` already blocks HTTP requests.

Reply:
Indeed, even the (mozilla) documentation says "The `upgrade-insecure-requests` directive is evaluated before `block-all-mixed-content` and if it is set, the latter is effectively a no-op."

Correcting. Thanks!